### DISCUSSION TITLE
Is it possible to set a key to close all tabs but current.

### DISCUSSION THREAD

**Discussion by BBennettChiba** [wezterm/wezterm#5813] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20240701-070926-69686f45

### Ask your question!

I hope to be able to close all tabs but the current one but I'm running into trouble. I thought there was a closetab action but I can't find it. I thought of looping through the tabs and making it active and then CloseCurrentTab action but that didn't work.

unrelated but I was also wondering if it was possible to change a visual setting during the time the leader key is active. All I can do now is change the cursor. I'd like something more visually striking.

**Comment by xMonody** [comment, marked as answer]
```lua
{
    key = "q",
    mods = "SHIFT|ALT",
    action = wezterm.action_callback(function(win, _)
        local tab = win:active_tab()
        local activeTabId = tab:tab_id()
        local muxWin = win:mux_window()
        local tabs = muxWin:tabs()
        for _, t in ipairs(tabs) do
            if t:tab_id() ~= activeTabId then
                t:activate()
                for _, p in ipairs(t:panes()) do
                    win:perform_action(wezterm.action.CloseCurrentPane({ confirm = false }), p)
                end
            end
        end
    end),
},
```

**Reply by BBennettChiba** [reply]
thank you. It works.

**Comment by xMonody** [comment]
If there are multiple panes in other windows, there seems to be an issue. Change 'CloseCurrentPane' to 'CloseCurrentTab' to fix it.
